File: goals.cpp
#include <iostream>
#include <cmath>
#include "convert.h"
#include "flight.h"
#include "goals.h"

void do_goals_calculations(double v0,double y0) {
	y0=tometers(y0);
	std::cout.precision(3);
	std::cout<<"DEG\tLOW\tMID\tHIGH"<<std::endl;
	for(double a=LOW_ANGLE;a<=HIGH_ANGLE;a+=ANGLE_STEP) {
		double vx0=v0*std::cos(torad(a));
		double vy0=v0*std::sin(torad(a));
		std::cout<<a<<"°\t";
		std::cout<<tofeet(simulate_flight(vx0,vy0,a,DELTAT,tometers(LOW_TARGET-y0)))<<"\t";
		std::cout<<tofeet(simulate_flight(vx0,vy0,a,DELTAT,tometers(MID_TARGET-y0)))<<"\t";
		std::cout<<tofeet(simulate_flight(vx0,vy0,a,DELTAT,tometers(HIGH_TARGET-y0)))<<std::endl;
	}
}

void do_find_velocity(double angle,double _dist,double y0) {
	std::cout.precision(3);
	std::cout<<"V0\tDIST"<<std::endl;
	for(double i=20.0;i<100.0;i+=0.1) {
		double vx0=tometers(i)*std::cos(torad(angle));
		double vy0=tometers(i)*std::sin(torad(angle));
		std::cout<<i<<"\t";
		double dist=tofeet(simulate_flight(vx0,vy0,angle,DELTAT,tometers(MID_TARGET-y0)));
		std::cout<<dist<<std::endl;
	}
}

File: java/Goals.java
public class Goals
{
	public static final double LOW_TARGET = 31.0;
	public static final double MID_TARGET = 793.0 / 8.0;
	public static final double HIGH_TARGET = 881.0 / 8.0;

	public static final double LOW_ANGLE = 0.0;
	public static final double HIGH_ANGLE = 60.0;
	public static final double ANGLE_STEP = 1.0;

	public static final double DELTAT = 0.001;

	public static void do_goals_calculations(double v0)
	{
		System.out.print("DEG\tLOW\tMID\tHIGH");
		System.out.print("\n");
		for (double a = LOW_ANGLE;a <= HIGH_ANGLE;a += ANGLE_STEP)
		{
			double vx0 = v0 * Math.cos(Convert.torad(a));
			double vy0 = v0 * Math.sin(Convert.torad(a));
			System.out.print(a);
			System.out.print("\t");
			System.out.print(Convert.tofeet(Flight.simulate_flight(vx0, vy0, a, DELTAT, Convert.tometers(LOW_TARGET))));
			System.out.print("\t");
			System.out.print(Convert.tofeet(Flight.simulate_flight(vx0, vy0, a, DELTAT, Convert.tometers(MID_TARGET))));
			System.out.print("\t");
			System.out.print(Convert.tofeet(Flight.simulate_flight(vx0, vy0, a, DELTAT, Convert.tometers(HIGH_TARGET))));
			System.out.print("\n");
		}
	}
}